src/tools/useElementScroll.ts

```diff
@@ -0,0 +1,83 @@
+import { throttle } from "lodash";
+import { useCallback, useEffect, useState } from "react";
+
+export type Position = Record<"x" | "y", number>;
+
+function getPosition<T extends HTMLElement>(
+  anchor?: T | null
+): Position | undefined {
+  if (!!anchor) {
```

Comment:
Unnecessary casting `anchor` to boolean inside conditionals throughout this entire file
```suggestion
  if (anchor) {
```